sbws/lib/relaylist.py

```diff
@@ -129,16 +195,38 @@ class RelayList:
     transparently in the background. Provides useful interfaces for getting
     only relays of a certain type.
     '''
-    REFRESH_INTERVAL = 300  # seconds
 
-    def __init__(self, args, conf, controller):
+    def __init__(self, args, conf, controller,
+                 measurements_period=MEASUREMENTS_PERIOD):
         self._controller = controller
         self.rng = random.SystemRandom()
         self._refresh_lock = Lock()
+        # To track all the consensus seen.
+        self._consensus_timestamps = []
+        # Initialize so that there's no error trying to access to it.
+        # In future refactor, change to a dictionary, where the keys are
+        # the relays' fingerprint.
+        self._relays = []
+        self._measurements_period = measurements_period
         self._refresh()
 
     def _need_refresh(self):
-        return time.time() >= self._last_refresh + self.REFRESH_INTERVAL
+        # New consensuses happen every hour.
+        return datetime.utcnow() >= \
+            self.last_consensus + timedelta(seconds=60*60)
+
+    @property
+    def last_consensus(self):
```

Comment:
I suggest naming this last_consensus_received_at or something like that -- this property sounds like it should return the last consensus itself.